code/app/src/main/java/com/cmput301w21t06/crowdfly/Views/ViewTrialLogActivity.java

```diff
@@ -389,4 +413,24 @@ public void onDoneGetExperimenterIds(ArrayList<String> ids){
         dropAdapter = new DropdownAdapter(this, R.layout.general_content,ids);
         dropdown.setAdapter(dropAdapter);
     }
+
+    @Override
+    public boolean onNavigationItemSelected(@NonNull MenuItem item) {
+        switch (item.getItemId()){
+
+            case R.id.hamHome:
+                Intent intent1 = new Intent(getApplicationContext(), MainActivity.class);
+                startActivity(intent1);
+                break;
+            case R.id.hamAccount:
+                Intent intent = new Intent(getApplicationContext(), UserProfileActivity.class);
+                intent.putExtra(TAG, UserController.reverseConvert(currentUser.getUserID()));
+                startActivity(intent);
+                break;
+            case R.id.hamExperiment:
+                break;
+
+        }
+        return true;
+    }
```

Comment:
This code should be put into an interface as a static method and just called in the listener rather than adding 20 lines of code in every activity

Reply:
I Ean lines 417-435 here